### PR TITLE
Handle file loading from "non-mod-rs" files properly

### PR DESCRIPTION
**Pull request by powerboat9**
This seems to address many of the remaining ```unknown macro``` errors from #2186

https://doc.rust-lang.org/reference/items/modules.html

https://dev-doc.rust-lang.org/beta/edition-guide/rust-2018/module-system/path-clarity.html#no-more-modrs

### CODE REVIEW

**Review by CohenArthur — COMMENTED** [review]
The changes seem okay to me, but they introduce a lot of test failures

**Review by philberty — COMMENTED** [review]
This LGTM but not 100% sure i can see whats going on with the test cases, why did they need to be moved around like this?

**Review by P-E-P — APPROVED** [review]

File: gcc/rust/ast/rust-ast.cc [hunk 1]
```diff
@@ -3310,21 +3310,40 @@ Module::process_file_path ()
...
+    {
+      including_fname = including_fpath.substr (dir_slash_pos + 1);
+    }
+
+  auto path_string = filename_from_path_attribute (get_outer_attrs ());
+
+  if (!path_string.empty () || !module_scope.empty ()
+      || including_fname == "mod.rs" || including_fname == "lib.rs"
+      || including_fname == "main.rs" || including_fname.size () < 3
+      || including_fname.compare (including_fname.size () - 3, 3, ".rs"))
```
Comment (CohenArthur) [inline gcc/rust/ast/rust-ast.cc:3340]: this condition is very big - why is the size of `including_fname` a determining factor here? also, can this condition be moved in an outer static function?

File: gcc/rust/parse/rust-parse-impl.h [hunk 2]
```diff
@@ -2430,8 +2430,27 @@ Parser<ManagedTokenSource>::parse_module (AST::Visibility vis,
...
+	    auto slash_idx = filename.rfind (file_separator);
+	    if (slash_idx == std::string::npos)
+	      slash_idx = 0;
+	    else
+	      slash_idx++;
+	    filename = filename.substr (slash_idx);
+
+	    if (filename != "mod.rs" && filename != "lib.rs"
+		&& filename != "main.rs" && filename.size () >= 3
+		&& !filename.compare (filename.size () - 3, 3, ".rs"))
```
Comment (CohenArthur) [inline gcc/rust/parse/rust-parse-impl.h:2447]: same question here

**Review by CohenArthur — APPROVED** [review]
That looks good to me - just the two comments to address and I'm okay with merging this. Thank you for working on this!

**Pull request merged**